yt_dlp/extractor/bilibili.py

```diff
@@ -1861,6 +1862,42 @@ def _real_extract(self, url):
             ie=BiliBiliIE.ie_key(), video_id=video_id)
 
 
+class BiliBiliDynamicIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:t\.bilibili\.com|(?:www\.)?bilibili\.com/opus)/(?P<id>\d+)'
+    _TESTS = [{
+        'url': 'https://t.bilibili.com/998134289197432852',
+        'info_dict': {
+            'id': 'BV1TAmBYVEJr',
+            'ext': 'mp4',
+            'uploader_id': '1192648858',
+            'comment_count': int,
+            '_old_archive_ids': ['bilibili 113457567568273_part1'],
+            'thumbnail': 'http://i2.hdslb.com/bfs/archive/50091efd965d9f13ff6814f7ad374f90ab21e77d.jpg',
+            'duration': 929.238,
+            'upload_date': '20241110',
+            'uploader': '何同学工作室',
+            'like_count': int,
+            'view_count': int,
+            'title': '美国小朋友就玩这个？！何同学工作室11月开箱',
+            'description': '本期产品信息：\n机器狗\n气味模拟器\nCloudboom Strike LS\n无弦吉他\n蓝牙磁带音箱\n神奇画板',
+            'timestamp': 1731232800,
+            'tags': list,
+        },
+    }]
+
+    def _real_extract(self, url):
+        t_id = self._match_id(url)
+        video_url = traverse_obj(self._download_json(
+            'https://api.bilibili.com/x/polymer/web-dynamic/v1/detail', t_id, query={
+                'id': t_id,
+            }, headers={
+                'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36',
+            }), ('data', 'item', ((), 'orig'), 'modules', 'module_dynamic', (('major', 'archive'), ('additional', 'reserve')), 'jump_url', any, {sanitize_url}))
```

Comment:
Support traversal of bangumi ones as well?
https://t.bilibili.com/1010093853244391443
https://t.bilibili.com/1007774564482023446
and example of link within the rich text content
https://t.bilibili.com/1005193024981958661

Reply:
will look into it when i have time. not sure if the links inside the rich text content should be extracted. do similar extractors like twitter's handle links in text like this?
i think we can put the rich text content into the playlist description instead. not sure if users can use --parse-metadata alongside additional_urls then

Reply:
bangumi patch:
```diff
diff --git a/yt_dlp/extractor/bilibili.py b/yt_dlp/extractor/bilibili.py
index daf89b20d..7204d3644 100644
--- a/yt_dlp/extractor/bilibili.py
+++ b/yt_dlp/extractor/bilibili.py
@@ -1892,7 +1892,9 @@ def _real_extract(self, url):
                 'id': t_id,
             }, headers={
                 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36',
-            }), ('data', 'item', ((), 'orig'), 'modules', 'module_dynamic', (('major', 'archive'), ('additional', 'reserve')), 'jump_url', any, {sanitize_url}))
+            }), ('data', 'item', ((), 'orig'), 'modules', 'module_dynamic',
+                 (('major', ('archive', 'pgc')), ('additional', ('reserve', 'common'))),
+                 'jump_url', any, {sanitize_url}))
         if not video_url:
             self.raise_no_formats('No video found!', expected=True, video_id=t_id)
         return self.url_result(video_url)
```